### PR TITLE
Update Ruff Rules and fix any problems it finds

### PR DESCRIPTION
**Pull request by nabobalis**

### CODE REVIEW

File: sunpy_soar/__init__.py [hunk 1]
```diff
@@ -1,6 +1,16 @@
-# Import here to register the client with sunpy
+"""
+``sunpy-soar``.
```
Comment (nabobalis) [inline sunpy_soar/__init__.py:2]:
```suggestion
``sunpy-soar``
```

File: .ruff.toml [hunk 2]
```diff
@@ -2,81 +2,58 @@
...
+    "ANN201", # Missing return type annotation for public function
+    "ANN202", # Missing return type annotation for private function
+    "ANN205", # Missing return type annotation for staticmethod
+    "ANN206", # Missing return type annotation for classmethod
     "COM812", # May cause conflicts when used with the formatter
-    "ISC001",  # May cause conflicts when used with the formatter
+    "D200", # One-line docstring should fit on one line
+    "D205", # 1 blank line required between summary line and description
+    "D401", # First line should be in imperative mood
+    "D404", # First word of the docstring should not be "This"\
```
Comment (nabobalis) [inline .ruff.toml:25]:
```suggestion
    "D400", # First line should end with a period
    "D401", # First line should be in imperative mood
    "D404", # First word of the docstring should not be "This"
```

**Review by hayesla — APPROVED** [review]
I'm not sure what the windows fails are, but if it passes all good

**Comment by nabobalis** [comment]
It is the MPL issue.

**Pull request merged**